File: leetcode/442/FindDuplicate.c
int *findDuplicates(int *nums, int numsSize, int *returnSize)
{
    int *list = (int *)malloc(numsSize * sizeof(int));
    int list_idx = 0;

    for (int i = 0; i < numsSize; i++)
    {
        int index = abs(nums[i]) - 1;
        if (nums[index] < 0)
        {
            list[list_idx++] = abs(nums[i]);
        }
        else
        {
            nums[index] = -nums[index];
        }
    }
    *returnSize = list_idx;
    return list;
}

File: leetcode/442/FindDuplicate.js
var findDuplicates = function(nums) {
    let list = [];
    
    for(let i = 0; i < nums.length; i++)
    {                
        let index = Math.abs(nums[i])-1;
        if(nums[index] < 0)
        {
            list.push(index+1);
        }else
        {
            nums[index] = -nums[index];
        }
    }
    return list;
};

let arr = [ 4, 3, 2, 7, 8, 2, 3, 1 ];
let result = findDuplicates(arr);
console.log(result);